### ISSUE TITLE
can't access emitted filename within `generateBundle`

### ISSUE THREAD

**Issue by danielroe** [rollup/rollup#4744]
### Rollup Version

3.7.0

### Operating System (or Browser)

MacOS

### Node Version (if applicable)

v18.12.1

### Link To Reproduction

https://github.com/danielroe/rollup-hash-repro

### Expected Behaviour

I expect the filename generated to be available within `generateBundle`.

**Note**: I didn't include the REPL.it link as it worked as expected there (with rollup v1.31.0) _until_ I upgraded the rollup version to 3.7.0 upon which it failed entirely with:

```
npx rollup --version & npx rollup -c
Cannot find module 'node:process'
Require stack:
- /home/runner/rollup-hash-repro/node_modules/rollup/dist/bin/rollup
exit status 1
```

### Actual Behaviour

```
❯ pnpm build

> rollup-hash-repro@1.0.0 build /Users/daniel/sandbox/rollup-hash-repro
> rollup -c rollup.config.js


src/index → stdout...
something.mjs-!~{001}~.js

//→ index.js:
const foo = 'bar';

export { foo };

//→ something.mjs-58e57b33.js:
var test = 'bobaloo';

export { test as default };
```

**Comment by lukastaegert** [comment]
Sorry about the REPL. It seems it still runs on Node 14. I really want to set up something new via StackBlitz but did not find the time.

In any case, `this.getFileName` should indeed work in generateBundle, I guess overlooked something here. Will have a look how I can fix this.

**Comment by lukastaegert** [comment]
Yes, completely forgot about this when implementing the new hashing. Fix at #4747.
